integration_tests/__tests__/toMatchSnapshot-test.js

```diff
@@ -50,26 +50,85 @@ test('basic support', () => {
 
   {
     const {stderr, status} = runJest(DIR, [filename]);
-    expect(status).toBe(0);
     expect(stderr).toMatch('1 snapshot written in 1 test file.');
+    expect(status).toBe(0);
   }
   {
     const {stderr, status} = runJest(DIR, [filename]);
-    expect(status).toBe(0);
     expect(stderr).toMatch('1 snapshot,');
+    expect(stderr).not.toMatch('1 snapshot written in 1 test file.');
+    expect(status).toBe(0);
   }
 
   makeTests({[filename]: template(['{apple: "updated value"}'])});
 
   {
     const {stderr, status} = runJest(DIR, [filename]);
-    expect(status).toBe(1);
     expect(stderr).toMatch('Received value does not match the stored snapshot');
+    expect(status).toBe(1);
   }
 
   {
     const {stderr, status} = runJest(DIR, [filename, '-u']);
-    expect(status).toBe(0);
     expect(stderr).toMatch('1 snapshot updated in 1 test file.');
+    expect(status).toBe(0);
+  }
+});
+
+test.skip('error thrown before snapshot', () => {
```

Comment:
this is the bug that is currently in master.
if an error is thrown before `toMatchSnapshot`, the snapshot will be treated as obsolete